src/client/App.jsx

```diff
@@ -14,16 +20,54 @@ function App() {
       .then(res => setMovies(res.data));
   }, []);
 
-  const handleRegister = async ({ username, password }) => {
-    
+  const handleRegister = async user => {
+    fetch(`${apiUrl}/user/register`, {
+      ...headers,
+      body: JSON.stringify(user)
+    })
+    .then(res => res.json())
+    .then(json => {
+      console.log(json)
+      if (json.error) {
+        window.alert(`${json.error}!`)
+        return
+      }
+      if (json.data.username) {
+        window.alert(`${json.data.username} registered!`)
+      }
+    })
   };
 
-  const handleLogin = async ({ username, password }) => {
-    
+  const handleLogin = async (user) => {
+    fetch(`${apiUrl}/user/login`, {
+      ...headers,
+      body: JSON.stringify(user)
+    })
+    .then(res => res.json())
+    .then(json => {
+      console.log(json)
+      if (json.error) {
+        window.alert(`${json.error}!`)
+        return
+      }
+      if (json.data) {
+        localStorage.setItem('token',json.data)
+        window.alert(`${json.data} token!`)
+      }
+    })
   };
   
-  const handleCreateMovie = async ({ title, description, runtimeMins }) => {
-    
+  const handleCreateMovie = async (movie) => {
```

Comment:
I like how you've constructed this handleCreateMovie function. The way you've simplified the fetch by creating the headersWithAuth variable is very neat.

Reply:
yeah I just didn't want to copy paste more fetch request boilerplate, this seems like the sort of thing someone could write a library/middleware for to tidy it up :)